sdk/head_tracker.cc

```diff
@@ -65,6 +65,28 @@ void HeadTracker::Resume() {
 void HeadTracker::GetPose(int64_t timestamp_ns,
                           std::array<float, 3>& out_position,
                           std::array<float, 4>& out_orientation) const {
+  Rotation rotation = GetPose(timestamp_ns);
+
+  out_orientation[0] = static_cast<float>(rotation.GetQuaternion()[0]);
+  out_orientation[1] = static_cast<float>(rotation.GetQuaternion()[1]);
+  out_orientation[2] = static_cast<float>(rotation.GetQuaternion()[2]);
+  out_orientation[3] = static_cast<float>(rotation.GetQuaternion()[3]);
+
+  out_position = ApplyNeckModel(out_orientation, 1.0);
+}
+
+void HeadTracker::Recenter() {
+  Rotation r = GetPose(0);
+  double yaw_angle = r.GetYawAngle();
+  recenter_rotation_ = recenter_rotation_ * Rotation::FromYawPitchRoll(-yaw_angle,0,0);
```

Comment:
Is it required to multiply the newly computed rotation by the previous one? Isn't enough just to save it? Please correct me if I'm wrong.

Reply:
It is required to multiply by the previous one for when a users makes a recenter more than one time. 
Let me explain: If I turn left 90 degrees from the center and recenter. `recenter_rotation_` will be a quaternion that makes a rotation of 90° in the y-axis direction. 
Now, if I go 45 more degrees to the left, `GetYawAngle()` will return 45°. But I need `recenter_rotation_` to make a rotation of 45° plus the other 90° of the previous recenter. So we multiply by the last recenter value to have the needed full 135° angle in the y-axis direction.

Reply:
Sounds good, thanks for the explanation!